lib/Parse/ParseStmt.cpp

```diff
@@ -107,15 +107,18 @@ bool Parser::isStartOfStmt() {
   }
 }
 
+// Checks the parser status of AST Node and parse the statemtnt.
 ParserStatus Parser::parseExprOrStmt(ASTNode &Result) {
+  // The token is ';'
+  // Check for error in syntax.
   if (Tok.is(tok::semi)) {
     SyntaxParsingContext ErrorCtxt(SyntaxContext, SyntaxContextKind::Stmt);
     diagnose(Tok, diag::illegal_semi_stmt)
       .fixItRemove(SourceRange(Tok.getLoc()));
     consumeToken();
     return makeParserError();
   }
-
+  // If the token '#' is at the start of line with a lead token 'code complition'.
```

Comment:
I’m not sure if the comments provide any benefits as the code is self-explanatory.

Reply:
I have done because every method was commented except this.